sourcery_analytics/extractors.py

```diff
@@ -122,26 +126,30 @@ def from_function(
         """Construct an arbitrary extractor from a function of a node."""
         return cls(FunctionVisitor(function))
 
-    @functools.singledispatchmethod
     def extract(self, item: Extractable) -> typing.Iterator[T]:
         """Extract from source code, a node, a file, or directory."""
+        # `singledispatchmethod` confuses mypy, so wrap with a mypy-friendly interface
```

Comment:
**suggestion (code_clarification):** Clarify the workaround for mypy and `singledispatchmethod`.

It's helpful to include a brief explanation or a link to documentation/issues that describe why `singledispatchmethod` is problematic with mypy. This context can aid future maintainers in understanding the decision and evaluating when it might be safe to remove the workaround.

```suggestion
        # Workaround: `singledispatchmethod` is not fully supported by mypy, leading to type checking issues.
        # This is a known limitation where mypy can struggle with dynamically generated methods like those created by `singledispatchmethod`.
        # We wrap the method with a mypy-friendly interface to ensure type safety and clarity.
        # For more details, see the relevant mypy issue: https://github.com/python/mypy/issues/2904
```